subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -173,11 +189,32 @@ public boolean isShowSinglePathToDependency() {
      *
      * @since 4.9
      */
-    @Option(option = "singlepath", description = "Show at most one path to each dependency")
+    @Option(option = "single-path", description = "Show at most one path to each dependency")
     public void setShowSinglePathToDependency(boolean showSinglePathToDependency) {
         this.showSinglePathToDependency = showSinglePathToDependency;
     }
 
+    @Internal
+    public boolean isShowingAllVariants() {
+        return showingAllVariants;
+    }
+
+    @Option(option = "all-variants", description = "Show all variants of each dependency")
+    public void setShowingAllVariants(boolean showingAllVariants) {
+        this.showingAllVariants = showingAllVariants;
+    }
+
+    @Internal
+    public Set<SelectionReasonOutput> getEnabledSelectionReasonOutputs() {
+        return enabledSelectionReasonOutputs;
+    }
+
+    @Option(option = "reasons", description = "Enable the given selection reason output")
+    public void setEnabledSelectionReasonOutputs(List<SelectionReasonOutput> enabledSelectionReasonOutputs) {
```

Comment:
You've already mentioned this below, but I think we should remove this option.

If you run into this in the future, we also have an `@OptionValues`, which can be used to control the set of values a command-line option can have.